aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUser.java

```diff
@@ -2332,6 +2334,24 @@ private void deleteAttributesInternal(final List<String> attributeNamesToDelete,
         cognitoIdentityProviderClient.deleteUserAttributes(deleteUserAttributesRequest);
     }
 
+    public void revokeTokens() {
+        CognitoUserSession cognitoUserSession = getCachedSession();
+        String accessToken = cognitoUserSession.getAccessToken().getJWTToken();
+        if (!CognitoJWTParser.hasClaim(accessToken, "origin_jti")) {
+            LOGGER.debug("Access Token does not contain `origin_jti` claim. Skip revoking tokens.");
+            return;
+        }
+        String refreshToken = cognitoUserSession.getRefreshToken().getToken();
+
+        RevokeTokenRequest request = new RevokeTokenRequest();
+        request.setToken(refreshToken);
+        request.setClientId(clientId);
+        if (!StringUtils.isBlank((clientSecret))) {
+            request.setClientSecret(clientSecret);
```

Comment:
Based on this logic, it looks revoking tokens will work whether you include `clientSecret` or not.  If that's the case, can we just not include `clientSecret` at all?

Reply:
Ahh, nevermind.  Per [the docs](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_RevokeToken.html#API_RevokeToken_RequestSyntax), it looks like it is required if the client id has a secret.